### ISSUE TITLE
Feature request: Allow for Horizontal  scrolling code blocks

### ISSUE THREAD

**Issue by moutikabdessabour** [r-lib/pkgdown#1447]
as pointed in #497 although the css was implemented in 03d93a1 but the R code that executes the examples doesn't allow for wide outputs. As it resets the console output width to 80 characters : https://github.com/r-lib/pkgdown/blob/6094ac3dd80a1f3f1d4e2f47d0ed4716d51362b0/R/build-reference.R#L177.
When this line is changed to allow for wider outputs :
```r
old_opt <- options(width = 10000)
```
The copy button gets misplaced when scrolling horizontally. 
![](https://i.imgur.com/zvSETAt.png)
I managed to circumvent this by changing the  83 line of the `pkgdown.js` file : https://github.com/r-lib/pkgdown/blob/6094ac3dd80a1f3f1d4e2f47d0ed4716d51362b0/inst/assets/pkgdown.js#L83
to :
```js
$(".examples, div.sourceCode").wrap("<div class='hasCopyButton'></div>");
```
basically wrapping the `pre` into `div` that'll be the parent to both the `button` and the `pre`.

**Comment by hadley** [comment]
If you want to use extra wide outputs in specific examples, you can use `options(width = 10000)` in that example. I don't think it's a good idea as a default.
